test/integration/test_raw_api.py

```diff
@@ -550,6 +551,71 @@ def raw_api_test_helper(raw_api, should_cleanup_old_buckets):
             is_file_lock_enabled=False,
         )
 
+    # b2_100_continue
+    print('b2_100_continue')
+    file_name = 'test-100-continue.txt'
+    file_contents = b'hello world'
+    file_sha1 = hex_sha1_of_stream(io.BytesIO(file_contents), len(file_contents))
+
+    def set_100_header(fn):
+        def wrapper(url, headers, *args, **kwargs):
+            headers.update({
+                'Expect': '100-continue',
+            })
+            return fn(url, headers, *args, **kwargs)
+
+        return wrapper
+
+    orig_send = http.client.HTTPConnection.send
+    sent_data = bytearray()
+
+    def patched_send(self, data):
+        sent_data.extend(data)
+        return orig_send(self, data)
+
+    with monkeypatch.context() as monkey:
+        monkey.setattr(
+            raw_api.b2_http,
+            'post_content_return_json',
+            set_100_header(raw_api.b2_http.post_content_return_json),
+        )
+        monkey.setattr(
+            http.client.HTTPConnection,
+            "send",
+            patched_send,
+        )
+        data = io.BytesIO(file_contents)
+
+        with pytest.raises(InvalidAuthToken):
+            raw_api.upload_file(
+                upload_url,
+                upload_auth_token + 'x',
+                file_name,
+                len(file_contents),
+                'text/plain',
+                file_sha1,
+                {'color': 'blue'},
+                data,
+                server_side_encryption=sse_b2_aes,
+            )
+
+        # Check if data was sent
+        assert file_contents not in sent_data
+
+        # this should not fail
+        data.seek(0)
+        raw_api.upload_file(
+            upload_url,
+            upload_auth_token,
+            file_name,
+            len(file_contents),
+            'text/plain',
+            file_sha1,
+            {'color': 'blue'},
+            data,
+            server_side_encryption=sse_b2_aes,
+        )
```

Comment:
please add `assert file_contents in sent_data` otherwise it very well may be that you are not patching the right function